pkg/services/rendering/http_mode.go

```diff
@@ -83,42 +113,81 @@ func (rs *RenderingService) renderViaHttp(ctx context.Context, renderKey string,
 		}
 	}()
 
+	var downloadFileName string
+	_, params, err := mime.ParseMediaType(resp.Header.Get("Content-Disposition"))
+	if err == nil {
+		downloadFileName = params["filename"]
+	}
+
+	err = rs.readFileResponse(resp, filePath, reqContext)
+	if err != nil {
+		return nil, err
+	}
+
+	return &RenderCSVResult{FilePath: filePath, FileName: downloadFileName}, nil
+}
+
+func (rs *RenderingService) doRequest(url *url.URL, headers map[string][]string, ctx context.Context) (*http.Response, error) {
+	req, err := http.NewRequest("GET", url.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("User-Agent", fmt.Sprintf("Grafana/%s", setting.BuildVersion))
+	for k, v := range headers {
+		req.Header[k] = v
+	}
+
+	req = req.WithContext(ctx)
+
+	rs.log.Debug("calling remote rendering service", "url", url)
+
+	// make request to renderer server
+	resp, err := netClient.Do(req)
+	if err != nil {
+		rs.log.Error("Failed to send request to remote rendering service.", "error", err)
+		return nil, fmt.Errorf("failed to send request to remote rendering service: %w", err)
+	}
+
+	return resp, nil
+}
+
+func (rs *RenderingService) readFileResponse(resp *http.Response, filePath string, ctx context.Context) error {
```

Comment:
By standard convention, `ctx context.Context` should be the first argument:
```suggestion
func (rs *RenderingService) readFileResponse(ctx context.Context, resp *http.Response, filePath string) error {
```